Automattic/jetpack
# Charts: update PieSemiCircleChart component

This PR refines the PieSemiCircleChart component with the following updates:
* **Removed `height` property**
  * The height property is unnecessary and has been removed to simplify the component’s API.
  * Component size is determined solely by the width, providing a more intuitive and straightforward approach.

* **Added `thickness` property**
  * Introduces flexibility in defining the pie’s “slice width” (thickness).
  * This allows customization of the chart’s visual appearance based on specific design requirements.

*  **Added `clockwise` property**
  * Adds the ability to configure the direction in which chart pieces are rendered.
  * This enables the display of pieces in the opposite direction to the default, enhancing adaptability for different use cases.

<img src="https://github.com/user-attachments/assets/22d077ba-43a2-49d7-942c-c595e176af42" width="300px " />

Fixes #

## Proposed changes:

*

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Run story book

```cli
cd projects/js-packages/charts
```

```cli
pnpm run storybook
```

2. Access the chart story

Open the following URL in your browser:
[http://localhost:50240/?path=/docs/js-packages-charts-types-pie-semi-circle-chart–docs](http://localhost:50240/?path=/docs/js-packages-charts-types-pie-semi-circle-chart--docs)

3. Interact with the controls
  * Use the provided Storybook controls to test the new features:
  * Check the behavior with the thickness property.
  * Toggle the clockwise property to observe its effect on the rendering direction.
  * Ensure the removal of the height property does not impact the component’s functionality.
  * Validate component behavior


https://github.com/user-attachments/assets/5491dec4-0e07-4c69-a98c-1bb093c3787e
